keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -758,6 +753,21 @@ _It is expected that the `.clusterset.local` zone is standard and available in
 all implementations, but customization and/or aliasing can be explored if there's
 demand._
 
+##### Not allowing cluster-specific targeting via DNS
+
+Both ClusterSetIP Services and Multicluster Headless Services are specified to explicitly disallow DNS records that target single clusters, such as using a formulation like `<clusterid>.<svc>.<ns>.svc.clusterset.local.` to target all 1+N backends in a single cluster.
+
+For ClusterSetIP services, this rationale is tied to the intent of its underlying ClusterIP Service. In a single-cluster setup, the purpose of a ClusterIP service is to reduce the context needed by the application to target ready backends, especially if those backends disappear or change frequently, and leverages kube-proxy to do this independent of the limitations of DNS. ([ref](https://kubernetes.io/docs/concepts/services-networking/service/#why-not-use-round-robin-dns)) Similarly, users of exported ClusterIP services should depend on the single `<clusterset-ip>` (or the single `A`/`AAAA` record mapped to it), instead of targeting per cluster backends. If a user has a need to target backends in a different way, they should use headless Services.
+
+For Multicluster Headless Services, the rationale is tied to the intent of its underlying Headless Service to provide absolutely no load balancing capabilities on any stateful dimension of the backends (such as cluster locality), and provide routing to each single backend for the application's purposes.
+
+In both cases, this restriction seeks to preserve the MCS position on [namespace sameness](https://github.com/kubernetes/community/blob/master/sig-multicluster/namespace-sameness-position-statement.md). Services of the same name/namespace exported in the multicluster environment are considered to be the same by definition, and thus their backends are safe to 'merge' at the clusterset level. If these backends need to be addressed differently based on other properties than name and namespace, they lose their fungible nature which the MCS API depends on. In these situations, those backends should instead be fronted by a Service with a different name and/or namespace.
```

Comment:
Ok, so if in any specific case, one of the clusters had a more-particular service, the architect would still have the chance to declare a secondary Service + ServiceExport (with the cluster name) targeting the same pods using the same selector no the service.

Reply:
Yes! :+1: (And just linking again for posterity/future readers of this comment thread -- this was discussed IRL in [SIG-Multicluster's 3/16/2021 meeting starting 36:28](https://youtu.be/6mqE8MDiSnQ?t=2188)).